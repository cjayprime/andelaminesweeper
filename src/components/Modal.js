import React, {Component} from 'react';
import {StatusBar, TouchableOpacity, View} from 'react-native';

import Animation from './Animation';

import * as Style from '../assets/styles';

export default class Modal extends Component {
  render() {
    const {onClose, children, animate} = this.props;
    return (
      <>
        <StatusBar hidden={true} />
        <TouchableOpacity
          onPress={onClose}
          style={Style.modal.modalContainer}
        />
        {animate ? (
          <Animation>{children}</Animation>
        ) : (
          <View style={Style.modal.popover}>{children}</View>
        )}
      </>
    );
  }
}
