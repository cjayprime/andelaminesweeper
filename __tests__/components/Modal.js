import React from 'react';
import {View} from 'react-native';

import Modal from '../../src/components/Modal';

import renderer from 'react-test-renderer';

jest.useFakeTimers();
describe('Modal Test', () => {
    test('Stability', () => {
        const tree = renderer.create(
            <Modal
                onClose={() => {}}
                animate={false}
            ><View></View></Modal>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});