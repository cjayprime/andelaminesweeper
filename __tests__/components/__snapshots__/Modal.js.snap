// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Test Stability 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000",
        "elevation": 100,
        "height": "100%",
        "left": 0,
        "opacity": 0.5,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 100000000000,
      }
    }
  />,
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 10,
        "elevation": 100,
        "left": "20%",
        "padding": 5,
        "position": "absolute",
        "top": "30%",
        "width": "60%",
        "zIndex": 50000000000000000,
      }
    }
  >
    <View />
  </View>,
]
`;
